x/gamm/types/gov_test.go

```diff
@@ -0,0 +1,75 @@
+package types_test
+
+import (
+	"testing"
+
+	proto "github.com/gogo/protobuf/proto"
+	"github.com/stretchr/testify/require"
+
+	"github.com/osmosis-labs/osmosis/v14/x/gamm/types"
+)
+
+func TestUpdateMigrationRecordsProposalMarshalUnmarshal(t *testing.T) {
+	tests := []struct {
+		proposal *types.UpdateMigrationRecordsProposal
+	}{
+		{ // empty title
+			proposal: &types.UpdateMigrationRecordsProposal{
+				Title:       "",
+				Description: "proposal to update migration records",
+				Records:     []types.BalancerToConcentratedPoolLink(nil),
+			},
+		},
+		{ // empty description
+			proposal: &types.UpdateMigrationRecordsProposal{
+				Title:       "title",
+				Description: "",
+				Records:     []types.BalancerToConcentratedPoolLink(nil),
+			},
+		},
+		{ // empty records
+			proposal: &types.UpdateMigrationRecordsProposal{
+				Title:       "title",
+				Description: "proposal to update migration records",
+				Records:     []types.BalancerToConcentratedPoolLink(nil),
+			},
+		},
+		{ // one record
+			proposal: &types.UpdateMigrationRecordsProposal{
+				Title:       "title",
+				Description: "proposal to update migration records",
+				Records: []types.BalancerToConcentratedPoolLink{
+					{
+						BalancerPoolId: 1,
+						ClPoolId:       5,
+					},
+				},
+			},
+		},
+		{ // two records
+			proposal: &types.UpdateMigrationRecordsProposal{
+				Title:       "title",
+				Description: "proposal to update migration records",
+				Records: []types.BalancerToConcentratedPoolLink{
```

Comment:
Would it be possible to test error case here by passing on invalid(wrong) proposal?

Reply:
Hey, so the only thing here that we are testing is proto marshaling and unmarshaling into the UpdateMigrationRecordsProposal struct. IMO testing a failing case would just prove that proto.Marshal and proto.Unmarshal works which shouldn't be necessary. Lmk if you disagree and I will find a way to test this

Reply:
I agree with you on this note! Resolving this